### PR TITLE
ocrvs-2739 Improved form draft fetching reducer

### PR DESCRIPTION
**Pull request by sadmananik**

### CODE REVIEW

File: packages/client/src/forms/configuration/reducer.ts [hunk 1]
```diff
@@ -71,6 +71,27 @@ export const formDraftReducer: LoopReducer<
...
+    case actions.GET_OFFLINE_DATA_SUCCESS: {
+      const offlineDataString = action.payload
+      const offlineData: IFormDraftData = JSON.parse(
+        offlineDataString ? offlineDataString : '{}'
+      )
+
+      if (isEmpty(offlineData)) {
+        return loop(state, Cmd.action(actions.fetchDraft()))
+      }
+      return state
```
Comment (Zangetsu101) [inline packages/client/src/forms/configuration/reducer.ts:91]: We need to return a new state with the `offlineData` that was loaded from indexedDB

File: packages/client/src/forms/configuration/reducer.ts [hunk 2]
```diff
@@ -40,8 +40,8 @@ export interface IDraft {
 }
 
 export interface IFormDraftData {
-  birth: IDraft
-  death: IDraft
+  birth: IDraft | null
+  death: IDraft | null
```
Comment (Zangetsu101) [inline packages/client/src/forms/configuration/reducer.ts:44]: I think we'll need to have discussion with @euanmillar about if we want to have default drafts in the db

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: packages/client/src/forms/configuration/reducer.ts (hunk 1 above)
Comment (sadmananik) [reply packages/client/src/forms/configuration/reducer.ts:91]: Done

File: packages/client/src/forms/configuration/reducer.ts (hunk 2 above)
Comment (sadmananik) [reply packages/client/src/forms/configuration/reducer.ts:44]: Yes. I will labeled the **Questions** tag in this PR.

Comment (Zangetsu101) [reply packages/client/src/forms/configuration/reducer.ts:44]: We will need default drafts to be available

**Pull request merged**